src/providers/dataProvider/tests/batch.test.ts

```diff
@@ -0,0 +1,217 @@
+import {
+  withLifecycleCallbacks,
+  type DataProvider,
+  type AuthProvider
+} from 'react-admin'
+import { R_AUDIT, R_BATCHES, R_USERS } from '../../../constants'
+import { DateTime } from 'luxon'
+import localForageDataProvider from 'ra-data-local-forage'
+import { lifecycleCallbacks } from '..'
+import { trackEvent } from '../../../utils/audit'
+import authProvider from '../../authProvider'
+import { encryptedUsers } from '../../../utils/init-data'
+import { AuditType } from '../../../utils/activity-types'
+import { generateDummyBatchForTesting } from './dummy-data'
+
+const TEST_STORAGE_KEY = 'rco-test'
+
+describe('CRUD operations on Batch Resource', () => {
+  let provider: DataProvider
+  let auth: AuthProvider
+  beforeAll(async () => {
+    const provider = await localForageDataProvider({
+      prefixLocalForageKey: TEST_STORAGE_KEY
+    })
+    for (const user of encryptedUsers) {
+      await provider.create<User>(R_USERS, { data: { ...user } })
+    }
+    auth = authProvider(provider)
+    await auth.login({ username: 'ian', password: process.env.PASSWORD })
+  })
+
+  beforeEach(async () => {
+    const withOutLifecycleProvider = await localForageDataProvider({
+      prefixLocalForageKey: TEST_STORAGE_KEY
+    })
+    provider = withLifecycleCallbacks(
+      withOutLifecycleProvider,
+      lifecycleCallbacks(
+        trackEvent(withOutLifecycleProvider),
+        withOutLifecycleProvider
+      )
+    )
+    // note: next block to be refactored, so we
+    // declare array of resource names, then loop through them,
+    // clearing each resource
+    const list = await provider.getList<Batch>(R_BATCHES, {
+      sort: { field: 'id', order: 'ASC' },
+      pagination: { page: 1, perPage: 1000 },
+      filter: {}
+    })
+    if (list.total !== undefined && list.total > 0) {
+      await provider.deleteMany(R_BATCHES, {
+        ids: list.data.map((item) => item.id)
+      })
+    }
+
+    const auditList = await provider.getList<Audit>(R_AUDIT, {
```

Comment:
@TahaKhanAbdalli - we should be using your generic clear method here in this class.